lang/java/reef-applications/reef-vortex/src/main/java/org/apache/reef/vortex/driver/DefaultVortexMaster.java

```diff
@@ -84,6 +86,40 @@
   }
 
   /**
+   * Add aggregate-able Tasklets to pendingTasklets.
+   */
+  @Override
+  public <TInput, TOutput> VortexAggregateFuture<TInput, TOutput>
+      enqueueTasklets(final VortexAggregateFunction<TOutput> aggregateFunction,
+                      final VortexFunction<TInput, TOutput> vortexFunction, final List<TInput> inputs,
+                      final Optional<FutureCallback<AggregateResult<TInput, TOutput>>> callback) {
+    final int aggregateFunctionId = aggregateIdCounter.getAndIncrement();
+    aggregateFunctionRepository.put(aggregateFunctionId, aggregateFunction);
+    final Codec<TOutput> aggOutputCodec = aggregateFunction.getOutputCodec();
+    final List<Tasklet> tasklets = new ArrayList<>(inputs.size());
+    final Map<Integer, TInput> taskletIdInputMap = new HashMap<>(inputs.size());
+
+    for (int i = 0; i < inputs.size(); i++) {
+      taskletIdInputMap.put(taskletIdCounter.getAndIncrement(), inputs.get(i));
+    }
+
+    final VortexAggregateFuture vortexAggregateFuture =
+        callback.isPresent() ?
+        new VortexAggregateFuture(executor, taskletIdInputMap, aggOutputCodec, callback.get()) :
```

Comment:
nit: we could add brackets (`VortexAggregateFuture<>`) to remove the warning about the raw type.